core/modules/replica/Configuration.h

```diff
@@ -429,75 +430,610 @@ class Configuration: public ConfigurationIFace {
      * @throws std::invalid_argument If the new value of the parameter is 0.
      */
     static unsigned int setDatabaseTransactionTimeoutSec(unsigned int value);
 
-    // ------------------------------------------------
-    // -- Global parameters of the XROOTD connectors --
-    // ------------------------------------------------
-
-    /// @return the default mode for XROOTD reconnects.
+    /// @return the default mode for xrootd reconnects.
     static bool xrootdAllowReconnect() { return _xrootdAllowReconnect; }
 
     /**
      * Change the default value of a parameter defining a policy for handling
-     * automatic reconnects to the XROOTD server. Setting 'true' will enable
+     * automatic reconnects to the xrootd server. Setting 'true' will enable
      * reconnects.
-     *
      * @param value The new value of the parameter.
      * @return The previous value.
      */
     static bool setXrootdAllowReconnect(bool value);
 
-    /// @return The default timeout for connecting to the XROOTD servers.
+    /// @return The default timeout for connecting to the xrootd servers.
     static unsigned int xrootdConnectTimeoutSec() { return _xrootdConnectTimeoutSec; }
 
     /**
      * Change the default value of a parameter specifying the maximum interval
-     * of time (seconds) to wait for connecting to the XROOTD servers before
+     * of time (seconds) to wait for connecting to the xrootd servers before
      * giving up (should this be enabled by the corresponding reconnect policy).
-     *
      * @param value The new value of the parameter (must be strictly greater than 0).
      * @return The previous value.
      * @throws std::invalid_argument If the new value of the parameter is 0.
      */
     static unsigned int setXrootdConnectTimeoutSec(unsigned int value);
 
-protected:
+    // ----------------------------------------------------------------------
+    // The instance API.
+    // ----------------------------------------------------------------------
+
+    Configuration(Configuration const&) = delete;
+    Configuration& operator=(Configuration const&) = delete;
+    ~Configuration() = default;
+
+    /**
+     * Reload non-static parameters of the Configuration from the same source
+     * they were originally read before.
+     * @note If the object was initialed from an in-memory object then
+     *   the method will do noting.
+     */
+    void reload();
+
+    /**
+     * Reload non-static parameters of the Configuration from the given JSON object.
+     * @note If the previous state of the object was configured from a source having
+     *   a persistent back-end (such as MySQL) then the association with the backend
+     *   will be lost upon completion of the method.
+     * @param obj The input configuration parameters.
+     * @throw std::runtime_error If the input configuration is not consistent
+     *   with expectations of the application.
+     */
+    void reload(nlohmann::json const& obj);
```

Comment:
There is a big difference between the static factory method `static Configuration::Ptr load(json&)` to which you're referring and the member method `reload(json&)`. The former makes a new object (the Configuration service), which the latter updates the state of the service. The Configuration service is set up just once. And it can be updated later. Note that many threads may have/cache pointers to an existing service. Updating the service from the JSON object has been added/designed to be used at workers in a context of the planned mechanism of the "on-the-wire" reconfiguration of those by the Master Controller on major events such as changes to the workers (removing/adding) and databases (ingested/deleted), etc. Right now all workers read the configurations from the same MySQL database which complicates a synchronization within the highly distributed and non-deterministic environment. The addition was also made to prepare for the change in the master-workers protocol that would work better in Kubernetes. The current protocol was designed for the stable host/Docker-based environments (what we have at NCSA) in which the host worker machines are always there, have the fixed DNS entries, and those entries always exist in the nameservices. IN Kubernetes things are ore complicated as workers may not be present in DNS when the master is starting. So, a better model is to reverse the protocol and let workers know about a master and report themselves to the one on their startup.  This topic goes far beyond your question. We can always discuss it at a group meeting if you want. The only reason why I'm m explaining it here is to give a context for the change (the method added to the Configuration API).